sdk/core/core/src/az_token_credential.c

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <_az_token_credential.h>
+
+#include <az_url.h>
+
+#include <_az_cfg.h>
+
+// Being given
+// "https://NNNNNNNN.vault.azure.net/secrets/Password/XXXXXXXXXXXXXXXXXXXX?api-version=7.0", gives
+// back "https://vault.azure.net" (needed for authentication).
+AZ_NODISCARD az_result
+_az_token_credential_get_resource(az_span const request_url, az_span_builder * const p_builder) {
```

Comment:
This mechanism for calculating a scope from the service url will not work for all services. For instance storage allows users to vanity urls, and in this case the service endpoint won't contain the reasource url at all. Also, some services may require different scopes for different paths in the service. This is yet another reason why the scopes needed for authentication should be dictated by the service client, not the credential, and not the end user.

Reply:
You told me before. This function will be changed later send an empty request and get the correct scope.
In general, I am going to remove the resource URI-based resource auto-detection from the auth policy implementation once again, and going to make it client's responsibility to provide the right scope during the policy initialization. This would allow different endpoints to use different scopes.
There might be a helper function for the clients to detect the right scopes, and it is likely that I will write it, but it will be clients who will be calling it before initializing the auth policy, so that they will provide the initializer with the correct scope.